File: Leetcode 0487 Max Consecutive Ones II [medium] [array]/Max Consecutive Ones II [two-pointers].js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let cnt = 0;
    let max = 0;
    let cur = 0;
    let allowFlips = 1;
    for (let num of nums) {
        if (num !== 0) {
            cnt++;
        } else {
            if (allowFlips !== 0) {
                cur = cnt + allowFlips;
                cnt = 0;
            }
        }
        
        max = Math.max(max, cnt + cur);
    }
    
    return max;
};

File: Leetcode 0487 Max Consecutive Ones II [medium] [array]/Max Consecutive Ones II [two-pointers].java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = 0;
        int cnt = 0;
        int cur = 0;
        int allowFlips = 1;
        for (int num : nums) {
            if (num != 0) {
                cnt++;
            } else {
                if (allowFlips >= 1) {
                    cur = cnt + allowFlips;
                    cnt = 0;
                }
            }
            max = Math.max(max, cnt + cur);
        }
        
        return max;
    }
}